src/architecture/utilities/tests/test_signalProcessing.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "architecture/utilities/signalProcessing.h"
+#include <Eigen/Dense>
+#include "opencv2/core.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/highgui.hpp"
+#include <gtest/gtest.h>
+#include <random>
+
+const double testAccuracy = 1e-8;
+
+double calculateStandardDeviation(const std::vector<double>& input)
```

Comment:
I think I found where you got this from: https://www.geeksforgeeks.org/cpp-program-to-find-the-standard-deviation-of-array-elements/

I guess there isn't a c++ library that we have that does this?

Reply:
there is not!